pkg/build/archive.go

```diff
@@ -32,7 +32,7 @@ func Archive(manifest model.Manifest) error {
 	}
 
 	// We build archives for each arch. These contain the same thing except arch specific istioctl
-	for _, arch := range []string{"linux", "osx", "win"} {
+	for _, arch := range []string{"linux-amd64", "linux-armv7", "linux-arm64", "osx", "win"} {
```

Comment:
I believe you need to retain `linux` here, then we can merge this PR.

Next, we would merge PR: https://github.com/istio/istio/pull/22381

A follow-on PR would then remove `linux` from the string slice.